spec/models/donation_spec.rb

```diff
@@ -9,16 +9,27 @@
   it { is_expected.to validate_numericality_of(:amount).is_greater_than(0) }
 
   describe ".search" do
-    let(:smithie_black) { create(:donor, name: "Smithie Black") }
-    let(:bob_smith_white) { create(:donor, name: "Bob Smith-White") }
-    let(:john_smith) { create(:donor, name: "John Smith") }
-    let(:carl_grey) { create(:donor, name: "Carl Grey") }
+    before do
```

Comment:
If I remove the unnecessary factories from `donor`, I can rewrite the test like this:
```
before do
      @match_behinning = create(:donation, donor: create(:donor, name: "Smithie Black"))
      @match_middle = create(:donation, donor: create(:donor, name: "Bob Smith White"))
      @match_end = create(:donation, donor: create(:donor, name: "John Smith"))
      @no_match = create(:donation, donor: create(:donor, name: "Carl Grey"))
    end

    it { expect(described_class.search("smith")).to match_array [@match_behinning, @match_middle, @match_end] } 
```

It is currently passing, and failing if `@no_match` is added to the assertion or `@match_end` is removed unlike the test with `let`s.